src/coprocessor/statistics/analyze.rs

```diff
@@ -76,35 +77,18 @@ impl<'a> AnalyzeContext<'a> {
         }
     }
 
-    // handle_index is used to handle `AnalyzeIndexReq`,
-    // it would build a histogram of index values.
-    fn handle_index(mut self) -> Result<Vec<u8>> {
-        let req = self.req.take_idx_req();
-        let mut scanner = IndexScanExecutor::new_with_cols_len(
-            req.get_num_columns() as i64,
-            self.ranges,
-            self.snap,
-            self.statistics,
-        );
-        let mut hist = Histogram::new(req.get_bucket_size() as usize);
-        while let Some(row) = scanner.next()? {
-            let bytes = row.data.get_column_values();
-            hist.append(bytes);
-        }
-        let mut res = analyze::AnalyzeIndexResp::new();
-        res.set_hist(hist.into_proto());
-        let dt = box_try!(res.write_to_bytes());
-        Ok(dt)
-    }
-
     // handle_column is used to process `AnalyzeColumnsReq`
     // it would build a histogram for the primary key(if needed) and
     // collectors for each column value.
-    fn handle_column(mut self) -> Result<Vec<u8>> {
+    fn handle_column(&mut self) -> Result<Vec<u8>> {
```

Comment:
Could we define it as `fn handle_column(mut self, statistics: &mut Statistics)` ,  since we would move some value and this function `handle_column` cann't be recalled?
So do functions `handle_index` and `handle_request`